### DISCUSSION TITLE
Future outlook for StaxRip

### DISCUSSION THREAD

**Discussion by stax76** [staxrip/staxrip#574] [General]
I want to take care of mpv.net for some months, if possible retire from my staxrip maintainer role because I'm not using it any longer, I've talked about it with Dendraspis long before.

Building the theme feature was very difficult and the result is really great, so Dendraspis can now have admin rights to the project if he wishes them.

With such a great team nobody has to be worried that the projects goes in the wrong direction, I would like to thank everybody for the effort and good atmosphere here.

Now what I think could be improved in the future. It would probably be nice having a little website hosted by GitHub, if nobody else can contribute maybe I can do it some time, it's something I want to learn anyway.

And better docs would probably help people. There is a great German tutorial here:

https://encodingwissen.de/praxis/staxrip

This could be translated with deepl.com and added to the wiki, as far as I know it's generally welcomed by the author and permitted by the license:

https://encodingwissen.de/info

Author Brother John is also a staxrip contributor, in particular helped with anything aspect ratio related, and of course helped with the tutorial, it does not only cover staxrip but encoding in general with great detail.

**Comment by ghost** [comment]
@stax76 ... Very sad that you leave.
But really happy the taking over is made by @Dendraspis 

Good luck, we'll be seeing you on mpv.net ! ❤️

**Comment by Dendraspis** [comment]
Did not see that coming to be honest. And I don't know what to answer. 😕 

We, meaning @stax76 , @44vince44 , @JJKylee  and me, made a huge progress in the last months. We had about 40 issues. Sooooo many bugs were fixed and some new features make it easier and more fun to use StaxRip. At the moment wie have only 12 *Issues* and all of them are feature requests. It was a really good job. Now it seems StaxRip reached a point where we can rest a bit. Maybe creating some wiki pages and introductions for beginner?! But overall StaxRip is working well now!

StaxRip is very complex and there are so many parts I have never worked with.  Of course I can support you, @stax76 , as I did before, but I don't know if it can work without you. 🤔  As far as I understand you don't want to spend time improving StaxRip and you also want to stop creating beta versions via your dropbox, is that right? 😲 

I also have to step back a bit and take care of my own projects, so I cannot provide the same power and speed I did in the last 2 months.  😞 

> so Dendraspis can now have admin rights to the project if he wishes them.

It's your *baby* for about 20 years, you decide.

**Reply by stax76** [reply]
Also @Patman86 and @DJATOM working on tool issues at least I found very important.

There are a couple of reasons why I want to slow down and if possible retire, if the project does okay without me. Helping a little if necessary is always possible.

The biggest reason is I'm not encoding any longer, this is true since not only a few but for many years. I use mpv.net every day, often hours.

From a technical viewpoint mpv and mpv.net are more exciting than staxrip. staxrip is VB.NET and video encoding, last year a little C++ which I enjoyed after a ten-year hiatus and having forgotten most of it sadly. I like VB.NET, but after 20 years there isn't much opportunity to learn something new, today it's not a very popular language, MS has stopped improving it, sadly. I'm not encoding, so my interest in video encoding is fairly limited. The people here are great, but it's difficult to follow so much activity, often I don't know when to stop, something I need to think about.

With mpv and mpv.net I can do whatever I want, C, Lua, JavaScript, PowerShell, Python script host, C#, .NET Core/5, support Linux using Avalonia if I want to. Working with mpv and libmpv is a great experience, it has a great design, I've learned so much from it, didn't even like and understand the command line before my mpv journey. And there is also a very passionate mpv community in various places.

Maintaining staxrip the cooperation with most of the tool authors is great, in the early days MP4Box was my least favorite tool, it was ugly, and in recent years the MP4Box devs helped us with everything very kindly. But constantly updating all these tools if very, very boring, maybe the most difficult part for me, we are at 210 tools at the moment, this could be reduced of course, people can use older tools and update them self if necessary.

> made a huge progress in the last months. We had about 40 issues. Sooooo many bugs were fixed and some new features make it easier and more fun to use StaxRip. At the moment wie have only 12 Issues and all of them are feature requests. It was a really good job. Now it seems StaxRip reached a point where we can rest a bit. Maybe creating some wiki pages and introductions for beginner?! But overall StaxRip is working well now!

I agree it's in a comfortable position currently, if I wanted to continue now I think next I would translate the guide of Brother John with help of deepl.com.

> StaxRip is very complex and there are so many parts I have never worked with. Of course I can support you, @stax76 , as I did before, but I don't know if it can work without you. 🤔 As far as I understand you don't want to spend time improving StaxRip and you also want to stop creating beta versions via your dropbox, is that right?

I could slowly reduce my effort, so it's easier for you, and if there are times not much happening it's not the end of the world, wouldn't be the first time. I can keep uploading Betas if it's difficult for you, but would not update tools that often.

> I also have to step back a bit and take care of my own projects, so I cannot provide the same power and speed I did in the last 2 months.

Yes, we should all try to relax.

I've sent you an invitation for the StaxRip org with owner privileges, and I've reduced Revan's privileges from owner to member since he is inactive since 1-2 years.

**Reply by Patman86** [reply]
Oh, you mentioned it once, but that the time has already come ... I think Dendraspis is exactly the right successor! He's also there with sweat and blood and knows what he's doing. He could already integrate some new features. It is nice to read that you would still be available if we need your help. Thank you for transferring the rights, I'll try to get involved in the best possible way.

**Comment by JJKylee** [comment]
I'm really sad to hear this after my one week's absence.
But like @stax76 said, I believe @Dendraspis will be excellently doing on @stax76's behalf as he has been doing so far.
That said, I'm pretty sure @stax76 will also be around for help, right? 😅
Anyway, let's just stay healthy and cheer up y'all~! 😁

**Reply by ghost** [reply]
@JJKylee nice to have you back, you made us worry!!
Well don't worry about @stax76 because unlike what he said before, he's STILL WITH US AND VERY ACTIVE 🎉 🤣 🚀 
Although it made some confusion, we're happy about it.
Of course @Dendraspis is perfect for the job, but @Dendraspis + @stax76 is EVEN BETTER !
**As long as they synchronize their efforts and work TOGETHER.**  (right, @stax76 ? that is important)

**Reply by stax76** [reply]
I'm still around helping a bit with certain dev tasks, today I will start building the themed TaskDialog, I will have to read some GIT tutorials regarding branching first. After that maybe also look into theming the tab control.

But then I really want to get into mpv.net, having already a wishlist with features that are difficult to build, the first bigger task will be a much better command palette, that will be used for certain features like a playlist UI.
